src/main/java/com/andrewthesailor/cinemaTickets/service/impl/ReservationServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+package com.andrewthesailor.cinemaTickets.service.impl;
+
+import com.andrewthesailor.cinemaTickets.DAO.ReservationDAO;
+import com.andrewthesailor.cinemaTickets.exception.TicketException;
+import com.andrewthesailor.cinemaTickets.model.Jsons.ReservationDetails;
+import com.andrewthesailor.cinemaTickets.model.Jsons.ReservationRequestData;
+import com.andrewthesailor.cinemaTickets.model.Jsons.TicketData;
+import com.andrewthesailor.cinemaTickets.model.*;
+import com.andrewthesailor.cinemaTickets.service.*;
+import com.andrewthesailor.cinemaTickets.utils.StringUtil;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.NoResultException;
+import javax.transaction.Transactional;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+@AllArgsConstructor
+public class ReservationServiceImpl implements ReservationService {
+
+    final
+    ScreeningService screeningService;
+
+    final
+    SeatService seatService;
+
+    final
+    TicketTypeService ticketTypeService;
+
+    final
+    ReservationDAO reservationDAO;
+
+    final
+    TicketService ticketService;
+
+    final
+    StringUtil stringUtil;
+
+    @Override
+    @Transactional
+    public ReservationDetails processReservation(ReservationRequestData data, Long screeningId) throws TicketException {
+        Screening screening;
```

Comment:
Brzydko tak, jak to można zrobić lepiej?

Reply:
Mówisz o całej klasie, która jest ledwo oszlifowana i jest do refaktoru, np zmiana metod voidowych, czy o @Transactional na metodzie w momencie, gdzie chyba lepiej dać na klasę(dobrze kojarzę?)?

Reply:
Mowię o linijce 46.

Reply:
Strzelam, że pewnie wrzucenie wszystkiego w blok try{}, albo rzucenie Ticket Exception z niższego poziomu. Byłbym wdzięczny, gdybyś napisał jak to lepiej rozwiązać jeżeli źle myślę, to będzie kolejna rzecz, której się nauczę po dzisiaj.

Reply:
Można przenieść inicjalizację do innej funkcji :)

Reply:
A no tak, nie pomyślałem o tej opcji. W sumie mogłoby wpaść w ramach rozbijania tego servicu